Feature custom themes logic

ali-aljufairi opened the pull request:
## TLDR
Based on @jacob314 feedback I am make new pull request where it is just the logic of loading custom theme currently there is no changes to the UI its just logic of loading custom themes from user setting but also I need to update theme 
to handle logic of validating 


Please tell me the next steps and I would like first to confirm what does the logic look for you and after finalizing logic I would work on UI but its already ready in other branch but need to see if there will be like a big shift before opening that 

also I still like your idea of generating based on wallpaper and would love to do it but maybe in seprate pr after logic and ui is done

## Dive Deeper


Example of my user setting 
```json 
{
  "selectedAuthType": "oauth-personal",
  "customThemes": {
    "Sachou": {
      "type": "custom",
      "name": "Sachou",
      "Background": "#1E1E2E",
      "Foreground": "#CDD6F4",
      "LightBlue": "#ADD8E6",
      "AccentBlue": "#89B4FA",
      "AccentPurple": "#CBA6F7",
      "AccentCyan": "#FFFFFF",
      "AccentGreen": "#A6E3A1",
      "AccentYellow": "#F9E2AF",
      "AccentRed": "#F38BA8",
      "Comment": "#6C7086",
      "Gray": "#FFAB12",
      "GradientColors": [
        "#4796E4",
        "#847ACE",
        "#C3677F"
      ]
    },
    "Ahmed": {
      "type": "custom",
      "name": "Ahmed",
      "Background": "#1E1E2E",
      "Foreground": "#CDD6F4",
      "LightBlue": "#A6BFEF",
      "AccentBlue": "#89B4FA",
      "AccentPurple": "#CBA6F7",
      "AccentCyan": "#89DCEB",
      "AccentGreen": "#A6E3A1",
      "AccentYellow": "#F9E2AF",
      "AccentRed": "#F38BA8",
      "Comment": "#89DCEB",
      "Gray": "#6C7086",
      "GradientColors": [
        "#4796E4",
        "#847ACE",
        "#C3677F"
      ]
    }
}
```




## Reviewer Test Plan



## Testing Matrix

<!-- Before submitting please validate your changes on as many of these options as possible -->

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | x  | ❓  | ❓  |
| npx      | ❓  | ❓  | ❓  |
| Docker   | x  | ❓  | ❓  |
| Podman   | ❓  | -   | -   |
| Seatbelt | ❓  | -   | -   |

## Linked issues / bugs
All Related issues in this
#2533

jacob314 commented on packages/cli/src/config/settings.ts:
```diff
@@ -63,6 +65,8 @@ export interface Settings {
   // UI setting. Does not display the ANSI-controlled terminal title.
   hideWindowTitle?: boolean;
+
+  // New UI setting to hide tips
```
remove this spurious comment

jacob314 commented on packages/cli/src/ui/App.tsx:
```diff
@@ -700,7 +700,7 @@ const App = ({ config, settings, startupWarnings = [] }: AppProps) => {
                 thought={
                   streamingState === StreamingState.WaitingForConfirmation ||
-                  config.getAccessibility()?.disableLoadingPhrases
+                    config.getAccessibility()?.disableLoadingPhrases
```
please run the formatter to avoid whitespace only changes.

jacob314 commented on packages/cli/src/ui/components/ThemeDialog.tsx:
```diff
@@ -73,19 +89,75 @@ export function ThemeDialog({
-  const [focusedSection, setFocusedSection] = useState<'theme' | 'scope'>(
-    'theme',
-  );
+  const handleCustomThemeSave = (customTheme: CustomTheme, scope: SettingScope) => {
```
revert this method

jacob314 commented on packages/cli/src/ui/hooks/useThemeCommand.ts:
```diff
@@ -107,10 +110,92 @@ export const useThemeCommand = (
     [applyTheme, loadedSettings],
   );
 
+  const handleCustomThemeSave = useCallback(
```
revert this

jacob314 commented on packages/cli/src/ui/themes/theme-manager.ts:
```diff
@@ -51,19 +53,152 @@ class ThemeManager {
+  /**
+   * Unregisters a custom theme.
+   * @param themeName The name of the custom theme to unregister.
+   * @returns True if the theme was successfully unregistered, false otherwise.
```
nit: don't need unregister yet if we don't have save.

jacob314 commented on packages/cli/src/ui/themes/theme.ts:
```diff
@@ -339,3 +159,257 @@ export class Theme {
+ * @param type The base theme type to derive from.
+ * @returns A new custom theme configuration.
+ */
+export function createDefaultCustomTheme(name: string, type: 'light' | 'dark'): CustomTheme {
```
probably not needed yet unless we in the future decide to support custom themes

jacob314 commented on packages/cli/src/ui/components/ThemeDialog.tsx:
```diff
@@ -73,19 +89,75 @@ export function ThemeDialog({
+
+  // Add a handler for navigation
   useInput((input, key) => {
+    if (showCustomThemeEditor) return;
```
remove all logic related to a custom theme editor from this pull request. expect most of the code in ThemeDialog should be reverted from this pr as the theme dialog should just behave as before with custom themes added to the list.

jacob314 reviewed (commented):
looks directionally right. only issue is there is still some code related to editing themes. please remove that to get this down to just the minimal support for defining custom themes in settings.json and loading them in the theme manager. I do wonder whether we should also allow users to specify custom themes as separate json files instead of placing them inline in settings.json. That is a bit of a one way door decision as once we allow this format we will have to support it for a while for backwards compatability. Fundamentally I would expect users want to copy theme.json files they downloaded from the web rather than writing themes inline in their settings.json file.

ali-aljufairi commented:
/gemini review

jacob314 commented on packages/cli/src/config/settings.ts:
```diff
@@ -100,9 +103,20 @@ export class LoadedSettings {
+        ...(workspace.customThemes || {}),
+      },
+      mcpServers: {
+        ...(user.mcpServers || {}),
```
wow good catch that we weren't merging mcp servers correctly previously. please add a test for this to prevent regressions.

ali-aljufairi replied:
okay I will add test it for it

jacob314 commented on packages/cli/src/config/settings.ts:
```diff
@@ -63,6 +65,7 @@ export interface Settings {
 
   // UI setting. Does not display the ANSI-controlled terminal title.
   hideWindowTitle?: boolean;
+
```
nit: revert added whitespace line.

jacob314 commented on packages/cli/src/ui/App.tsx:
```diff
@@ -147,7 +147,7 @@ const App = ({ config, settings, startupWarnings = [] }: AppProps) => {
     handleThemeSelect,
     handleThemeHighlight,
-  } = useThemeCommand(settings, setThemeError, addItem);
+  } = useThemeCommand(settings, addItem, setThemeError);
```
this code changed even though no functionality was added. please revert back to minimize the diff.

ali-aljufairi replied:
For App.tsx  i just copied the lastest version at the time of working and it had that change at that I will revert

jacob314 commented on packages/cli/src/ui/components/ThemeDialog.tsx:
```diff
@@ -37,18 +37,27 @@ export function ThemeDialog({
   // Generate theme items
-  const themeItems = themeManager.getAvailableThemes().map((theme) => {
+  // Ensure custom themes are loaded before rendering
+  useEffect(() => {
```
this seems a bit odd. why is the theme dialog triggering the themeManager to load the custom themes. this should instead occur on startup loading the custom themes into the theme manager regardless of whether the theme dialog is loaded.

ali-aljufairi replied:
I reverted to ThemeDialog.tsx how main is and custom theme is still working

ali-aljufairi replied:
basically you show the file and copy it to your branch and you can copy it also in different to use it as refernce

jacob314 commented on packages/cli/src/ui/components/ThemeDialog.tsx:
```diff
@@ -37,18 +37,27 @@ export function ThemeDialog({
   const [selectInputKey, setSelectInputKey] = useState(Date.now());
 
-  // Determine which radio button should be initially selected in the theme list
+  // Determine which radio button should be initially selected in the themes list
```
revert this comment change

jacob314 commented on packages/cli/src/gemini.tsx:
```diff
@@ -123,6 +123,9 @@ export async function main() {
   }
 
+  // Load custom themes from settings
+  themeManager.loadCustomThemes(settings.merged.customThemes);
```
this is a good place to be loading the custom themes. I suspect the call to load in ThemeDialog wasn't needed.

ali-aljufairi replied:
Yup I have removed it from theme dialog

jacob314 commented on packages/cli/src/ui/themes/theme-manager.ts:
```diff
@@ -51,19 +59,121 @@ class ThemeManager {
     this.activeTheme = DEFAULT_THEME;
   }
 
+  /**
```
please add tests for this new code in theme manager

jacob314 commented on packages/cli/src/ui/hooks/useThemeCommand.ts:
```diff
@@ -96,9 +64,50 @@ export const useThemeCommand = (
     (themeName: string | undefined, scope: SettingScope) => {
-      // Added scope parameter
       try {
+        // If applying to user settings and the theme is a custom theme from workspace, copy its definition
```
I'm not sure we should do this. The user now ends up with a theme defined a place they didn't expect. I would prefer for us to keep this simple. If I select a theme the only exaction that makes sense is that I changed my selected theme not that I saved custom themes from one file to another. I would be more comfortable with users getting a theme not found error if they opened gemini cli in a different directory and the theme wasn't available.
alternately we could only surface themes defined at the user level when selecting a theme at the user level but I am not sure that is crucial.

jacob314 reviewed (commented):
Done with a round of review. I think this will be in a state I can approve after all the comments are addressed. Sorry for all the comments. This was a fairly complicated change to get right and there were a bunch of nuanced tradeoffs you needed to make.

ali-aljufairi commented on packages/cli/src/ui/components/ThemeDialog.tsx:
```diff
@@ -179,24 +222,33 @@ export function ThemeDialog({
+            initialIndex={selectedThemeIndex}
             onSelect={handleThemeSelect}
-            onHighlight={onHighlight}
+            onHighlight={(themeName) => {
```
so basically the difference it mainly because how the code is formated 

I run 

```
npm run format
```
which change the how the block of code is rendereder

and I only need to update  how to code is render based on active theme this what I am trying to do

ali-aljufairi replied:
Current main

![image](https://github.com/user-attachments/assets/c2e200eb-25b9-4d3d-8773-cad0da5bda23)



My update

![image](https://github.com/user-attachments/assets/71803f3a-dc23-48a6-afda-c96649d2f01d)


![image](https://github.com/user-attachments/assets/82445d6d-861e-4103-88f2-88b9c78d40b4)

ali-aljufairi commented:
1.	ThemeDialog.tsx
	•	Restored to the original version from main and re-applied only necessary changes.
	•	Main update: Theme preview now updates immediately after selecting a theme.
         there is  a problem that I have mentioned up that there is no of showing different lines 
     if the theme showed 
     
     
     I installed gemmin using npx 
     
     what I got 
     
![image](https://github.com/user-attachments/assets/28e91c88-620a-44b9-aa1c-63d3a0d765ff)

     My current update what it makes
     
![image](https://github.com/user-attachments/assets/bb665e1d-91d8-496f-8947-63d0baa33b72)

     
![image](https://github.com/user-attachments/assets/a9df3b7a-91c0-421d-b019-209347ba4f3d)

![image](https://github.com/user-attachments/assets/3fa11ea0-4a5b-4582-a902-64f474c44675)

	
	
	
	2.	useAuthCommand.ts
	•	Updated to fix a build issue by explicitly casting authMethod as AuthType.
	•	⚠️ Merge conflict occurred in this file:

<<<<<<< feature-custom-themes-logic
settings.setValue(scope, 'selectedAuthType', authMethod as AuthType);
=======
settings.setValue(scope, 'selectedAuthType', authType);
>>>>>>> main


	3.	settings.ts
	•	Refactored using a generic type as per your suggestion.
	•	This change contributed to the merge conflict above.
	4.	Empty Test Case Handling
	•	Modified so that merging returns empty objects for both customTheme and mcpTheme
	•	Let me know if this is the appropriate behavior.
        •   Also added  Test cases for this
	5.	Custom Theme Scope Filtering
	•	Custom themes are now filtered based on the selected user scope.
	•	Removed logic that copied all themes regardless of scope.
	6.	ThemeManager Tests
	•	Added test cases for ThemeManager as requested.
	7.	Whitespace Fixes
	•	Cleaned up minor formatting issues.

ali-aljufairi commented:
Also I have been wondering  how will user know  what to place in settings.json file should we include  it in the docs @jacob314

jacob314 commented:
+1 to updating our docs in a follow up. I believe we need much more extensive docs of settings.json as well as a `/settings` command that makes it clear to users what all the options are as well as basic functionality to edit them directly.

ali-aljufairi commented:
I would work on /settings but I need to dive deep more on the product and try to understand what is possible to edit

jacob314 commented on packages/cli/src/ui/utils/CodeColorizer.tsx:
```diff
@@ -100,9 +100,10 @@ export function colorizeCode(
 ): React.ReactNode {
   const codeToHighlight = code.replace(/\n$/, '');
-  const activeTheme = themeManager.getActiveTheme();
+  const activeTheme = theme || themeManager.getActiveTheme();
```
please revert all changes to this file. they are not related to this pull request topic.

jacob314 replied:
never mind. now I understand. this change makes sense.

jacob314 commented on packages/cli/src/ui/components/ThemeDialog.tsx:
```diff
@@ -36,16 +36,35 @@ export function ThemeDialog({
+  // Generate theme items filtered by selected scope
+  const customThemes =
+    selectedScope === SettingScope.User
+      ? settings.user.settings.customThemes || {}
```
same issue: too much filtering. the filter for user level themes is right but for workspace we should just show all themes.

ali-aljufairi replied:
Done

jacob314 reviewed (approved):
Thanks for sticking with this pull request. It now looks good. I'll to a small tiny review after the merge conflicts are resolved and tests pass.

jacob314 commented on packages/cli/src/ui/components/ThemeDialog.tsx:
```diff
@@ -45,7 +45,7 @@ export function ThemeDialog({
     selectedScope === SettingScope.User
       ? settings.user.settings.customThemes || {}
-      : settings.workspace.settings.customThemes || {};
+      : { ...(settings.user.settings.customThemes || {}), ...(settings.workspace.settings.customThemes || {}) };
```
you can just use the merged themes for this one.

ali-aljufairi replied:
yeah you are right

ali-aljufairi replied:
@jacob314  is there anything more to do or stuff to check ?

jacob314 commented:
you should accept the version I suggested. strange

> @jacob314
> 
> Sorry for asking I am not informed what should I do with the conflict from my side my changes should be accepted because I got the error based on fixed you informed me to do
> 
> so please inform how to solve the conflict should I revert the change you suggested or what The conflict is only one line
> 
> ```shell
> setValue<K extends keyof Settings>(
>     scope: SettingScope,
>     key: K,
>     value: Settings[K],
>   ): void {
>     const settingsFile = this.forScope(scope);
>     settingsFile.settings[key] = value;
>     this._merged = this.computeMergedSettings();
>     saveSettings(settingsFile);
>   }
> 
> 
> packages/cli/src/ui/hooks/useAuthCommand.ts
> ```

Remove the `as` if the code type checks without the as. Leave it in if the code useAuthCommand now requires the `as` to typecheck with your change.

ali-aljufairi commented:
@jacob314  its required for the code to compile and build and check 


if I remove I am unable to build

ali-aljufairi commented:
<img width="1832" alt="image" src="https://github.com/user-attachments/assets/3ae744c9-2bb5-4753-9fa8-b1d71a4e170c" />


this what happen if I `as`

ali-aljufairi commented:
<img width="753" alt="image" src="https://github.com/user-attachments/assets/3ec764ee-4858-4560-9759-63c775f91168" />
I am able to build if as AuthType

I am not sure if there other solution or better approach to the problem if there is pelase inform and will work on it 

@jacob314

jacob314 commented:
> <img alt="image" width="753" src="https://private-user-images.githubusercontent.com/85583841/462133350-3ec764ee-4858-4560-9759-63c775f91168.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE1NjI1NjEsIm5iZiI6MTc1MTU2MjI2MSwicGF0aCI6Ii84NTU4Mzg0MS80NjIxMzMzNTAtM2VjNzY0ZWUtNDg1OC00NTYwLTk3NTktNjNjNzc1ZjkxMTY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzAzVDE3MDQyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzODhlZmYxMzNiY2QzYmZiMjkxMTlhYzg5NDg2MzllMzJjYjY2NGFkM2U2ZTkyMzI5NjU2Zjc1NTI4N2YzNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CTNROTGFqRo4OqzQfTzxDJF2X8CTjmr3yvs1EpWLXns"> I am able to build if as AuthType
> I am not sure if there other solution or better approach to the problem if there is pelase inform and will work on it
> 
> @jacob314

using the as is alright. Lets leave cleaning that up further for a follow up pull request. The right solution will be to make more of the methods in useAuthCommand require the AuthType instead of accepting arbitrary strings. All that happened here is your improvement to settings.ts caught  type issue that was previously unnoticed because setValue didn't have strict enough types.

ali-aljufairi commented:
@jacob314  hope you are doing well is there somthing I can do to get this merged also I have experminted 

in both rebasing follow and merge follow both would result into merge conflict and and solution should be something like this

I merged my feature barnch with upstream/main and tested it everything is working but you need make this line as follow

![image](https://github.com/user-attachments/assets/b84c8593-aeed-4a87-8092-a73fb0e10edc)

Because My merge was not there I think someone have fixed the type issue and  now you can just make it like this 

dealying the merge would more trouble as time pass so pleas give a look in you free time 


and if you want me to do anything to fix it please inform me possible solution ?

ali-aljufairi commented:
#2721 
This link to the pull request that fix the issue there is approach is more clean and I think we should keep it as it is and he also removed unused methods

@jacob314

ali-aljufairi commented:
@scidomino


Hello I am unable to build the project due the fact for 

schemaVaildator.ts


#2881

The way you imported ajv

```js
import * as ajv from 'ajv';

const ajValidator = new ajv.Ajv();

```



The error I get when building locally I have multiple approaches to fix this the build which works locally with but when building here does not work any tips  or  guidencae 

### Current Code change 

I think this could pass the build 

```js
import Ajv from 'ajv';

const ajValidator = new Ajv();

```


### Local Build Error

```js


> @google/gemini-cli-core@0.1.9 build
> node ../../scripts/build_package.js

src/utils/schemaValidator.ts:10:29 - error TS2339: Property 'Ajv' does not exist on type '{ default: { (options?: Options | undefined): Ajv; new (options?: Options | undefined): Ajv; ValidationError: typeof ValidationError; MissingRefError: typeof MissingRefError; $dataMetaSchema: object; }; ValidationError: typeof ValidationError; MissingRefError: typeof MissingRefError; $dataMetaSchema: object; }'.

10 const ajValidator = new ajv.Ajv();
                               ~~~


Found 1 error.

node:internal/errors:983
  const err = new Error(message);
              ^

Error: Command failed: tsc --build
    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at checkExecSyncError (node:child_process:892:11)
    at execSync (node:child_process:964:15)
    at file:///Users/ali/github/gemini-cli/scripts/build_package.js:30:1
    at ModuleJob.run (node:internal/modules/esm/module_job:365:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:665:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:99:5) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 28680,
  stdout: null,
  stderr: null
}

Node.js v24.3.0
npm error Lifecycle script `build` failed with error:
npm error code 1
npm error path /Users/ali/github/gemini-cli/packages/core
npm error workspace @google/gemini-cli-core@0.1.9
npm error location /Users/ali/github/gemini-cli/packages/core
npm error command failed
npm error command sh -c node ../../scripts/build_package.js
node:internal/errors:983
  const err = new Error(message);
              ^

Error: Command failed: npm run build --workspaces
    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at checkExecSyncError (node:child_process:892:11)
    at execSync (node:child_process:964:15)
    at file:///Users/ali/github/gemini-cli/scripts/build.js:35:1
    at ModuleJob.run (node:internal/modules/esm/module_job:365:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:665:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:99:5) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 28607,
  stdout: null,
  stderr: null
}

Node.js v24.3.0
```

ali-aljufairi commented:
@jacob314  @google-gemini/gemini-cli-askmode-approvers get approvale 

is it possible to start testing and building because I can run on my local machine and it will be okay and I run and build docker is fine so can you please give approval

jacob314 commented:
Thanks so much. Unfortunately there are still at least four failing tests. You should be able to reproduce the failures locally. If not please let me know. Failures I see on the bots:
```
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 4 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/ui/themes/theme-manager.test.ts > ThemeManager > should set and get the active theme
AssertionError: expected 'NoColor' to be 'Default' // Object.is equality

Expected: "Default"
Received: "NoColor"

 ❯ src/ui/themes/theme-manager.test.ts:50:48
     48| 
     49|   it('should set and get the active theme', () => {
     50|     expect(themeManager.getActiveTheme().name).toBe(DEFAULT_THEME.name…
       |                                                ^
     51|     themeManager.setActiveTheme('Ayu');
     52|     expect(themeManager.getActiveTheme().name).toBe('Ayu');

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/4]⎯

 FAIL  src/ui/themes/theme-manager.test.ts > ThemeManager > should set and get a custom active theme
AssertionError: expected 'NoColor' to be 'MyCustomTheme' // Object.is equality

Expected: "MyCustomTheme"
Received: "NoColor"

 ❯ src/ui/themes/theme-manager.test.ts:58:48
     56|     themeManager.loadCustomThemes({ MyCustomTheme: validCustomTheme });
     57|     themeManager.setActiveTheme('MyCustomTheme');
     58|     expect(themeManager.getActiveTheme().name).toBe('MyCustomTheme');
       |                                                ^
     59|   });
     60| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/4]⎯

 FAIL  src/ui/themes/theme-manager.test.ts > ThemeManager > should return false when setting a non-existent theme
AssertionError: expected 'NoColor' to be 'Default' // Object.is equality

Expected: "Default"
Received: "NoColor"

 ❯ src/ui/themes/theme-manager.test.ts:63:48
     61|   it('should return false when setting a non-existent theme', () => {
     62|     expect(themeManager.setActiveTheme('NonExistentTheme')).toBe(false…
     63|     expect(themeManager.getActiveTheme().name).toBe(DEFAULT_THEME.name…
       |                                                ^
     64|   });
     65| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/4]⎯

 FAIL  src/ui/themes/theme-manager.test.ts > ThemeManager > should fallback to default theme if active theme is invalid
AssertionError: expected 'NoColor' to be 'Default' // Object.is equality

Expected: "Default"
Received: "NoColor"

 ❯ src/ui/themes/theme-manager.test.ts:88:48
     86|       type: 'custom',
     87|     };
     88|     expect(themeManager.getActiveTheme().name).toBe(DEFAULT_THEME.name…
       |                                                ^
     89|   });
     90| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/4]⎯


 Test Files  1 failed | 75 passed | 1 skipped (77)
      Tests  4 failed | 814 passed | 10 skipped (828)
   Start at  06:12:38
   Duration  47.70s (transform 3.18s, setup 0ms, collect 49.61s, tests 13.54s, environment 45.33s, prepare 8.23s)
   ```

ali-aljufairi commented:
@jacob314  Test is failing because the new package vscode-ide-compantion package still has not test in it so should I add it myself or this will be merged and fix later

jacob314 reviewed (approved):
![lgtm](https://i.imgur.com/JgNRalI.png)
thank you for sticking with this pr! Can't wait to see what users do with custom themes.
